### PR TITLE
Use core function to register Add to Cart with Options (experimental) block

### PR DESCRIPTION
**Pull request by retrofox**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

The main change in terms of implementation is using the registerBlockType() function to register the `Add To Cart with Options (Experimental)` block.
Using a custom register diverges from the core and is not aligned with the idea of being Gutenberg-friendly, which has largely talked about recently.

Also:
* It moves the template from the constant to the Edit component function. 
* Tweak a bit the block description (we can update it later)
* Move a block definition to `block.json` file

Closes # .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

No significant changes here in terms of functionallity

1. Ensure install and activate the `WooCommerce Beta Tester` plugin
image

2. Go to the `WooCommerce Admin Test Helper` -> `Features` page

image

3. Confirm the `experimental-blocks` and `blockified-add-to-cart` flags are enabled

Screenshot 2024-12-02 at 12 40 34 PM

4. Edit the `Single Product` template
5. Add the `Add To Cart with Options (Experimental)` block
Screenshot 2024-12-02 at 1 16 29 PM

6. Confirm the block works as expected

Screenshot 2024-12-02 at 1 14 52 PM

7. Create/edit a new post
8. Add a Single Product Block
9. Add the `Add To Cart with Options (Experimental)` block
10. Confirm it works as expected

Screenshot 2024-12-02 at 12 42 57 PM

11. Confirm that removing the feature flags the blocks are not available 

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [x] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/constants.ts [hunk 1]
```diff
@@ -1,14 +1,3 @@
...
-
-export const INNER_BLOCKS_TEMPLATE: InnerBlockTemplate[] = [
-	[
-		'woocommerce/product-button',
-		{
-			textAlign: 'center',
-			fontSize: 'small',
-		},
-	],
-];
```
Comment (octaedro) [inline plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/constants.ts:14]: ❤️

File: plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/index.tsx [hunk 2]
```diff
@@ -12,27 +12,10 @@ import metadata from './block.json';
...
-	},
-};
-
 if ( isExperimentalBlocksEnabled() ) {
-	registerBlockSingleProductTemplate( {
-		blockName: metadata.name,
-		blockMetadata: metadata,
-		blockSettings,
-		isAvailableOnPostEditor: true,
+	registerBlockType( metadata, {
```
Comment (octaedro) [inline plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/index.tsx:16]: I think that if we don’t register the block this way, it won’t be available in the `Single Product` template.

Screenshot 2024-12-02 at 12 27 14 PM

I'm okay with replacing the registering method, but as far as I understand, the `Add to Cart with Options (Experimental)` block should also be available for use in the `Single Product` template.

Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/index.tsx:16]: it should be there, in fact, it's described in the testing instructions

Comment (octaedro) [reply plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/index.tsx:16]: Sorry, what I tried to mean is that I do not see it available under the `Single Product` template.

https://github.com/user-attachments/assets/59792216-4689-4b51-b2e9-0223287683a6

Comment (retrofox) [reply plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/index.tsx:16]: Yes, you're right, Nice catch, Fernando. Let me update the code.

**Review by octaedro — APPROVED** [review]
Great job @retrofox! Thank you for addressing the change. This is testing well on my end and the code LGTM 🚀 

Tested:
- ✅ `Single Product` template
- ✅ `Single Product` block

**Pull request merged**